### DISCUSSION TITLE
Canceling Active Jobs

### DISCUSSION THREAD

**Discussion by jgalbsss** [taskforcesh/bullmq#2948] [Q&A]
I'm dealing with a tricky situation in our BullMQ implementation where jobs occasionally get stuck in the "active" state. We've seen this happen when our processors crash or timeout, but the jobs remain locked in Redis.

I've looked into BullMQ's cancellation features with observables, but I'm wondering if there is a free solution.

Specifically, I'm looking for:
- A way to gracefully cancel/fail these stuck jobs
- Methods to identify and clean up these zombie jobs
- Best practices for handling this in production without impacting other running jobs

Has anyone implemented a solution for this?

**Comment by manast** [comment]
BullMQ has a built in mechanism for detecting stalled jobs and move them back to wait for re-processing (by default one time but you can increase this setting if necessary): https://docs.bullmq.io/guide/jobs/stalled 
This would basically solve your issue, as this mechanism was designed specifically for handling the case where workers die unexpectedly.
